hs_geographic_feature_resource/receivers.py

```diff
@@ -388,96 +416,108 @@ def geofeature_pre_add_files_to_resource(sender, **kwargs):
-
-                    originalcoverage_dict = meta_dict["originalcoverage"]['originalcoverage']
-                    res_obj.metadata.create_element('OriginalCoverage',
-                                                    northlimit=originalcoverage_dict['northlimit'],
-                                                    southlimit=originalcoverage_dict['southlimit'],
-                                                    westlimit=originalcoverage_dict['westlimit'],
-                                                    eastlimit=originalcoverage_dict['eastlimit'],
-                                                    projection_string=originalcoverage_dict['projection_string'],
-                                                    projection_name=originalcoverage_dict['projection_name'],
-                                                    datum=originalcoverage_dict['datum'],
-                                                    unit=originalcoverage_dict['unit'])
-
-                    field_info_array = meta_dict["field_info_array"]
-                    for field_info in field_info_array:
-                        field_info_dict = field_info["fieldinformation"]
-                        res_obj.metadata.create_element('FieldInformation',
-                                                    fieldName=field_info_dict['fieldName'],
-                                                    fieldType=field_info_dict['fieldType'],
-                                                    fieldTypeCode=field_info_dict['fieldTypeCode'],
-                                                    fieldWidth=field_info_dict['fieldWidth'],
-                                                    fieldPrecision=field_info_dict['fieldPrecision'])
-
-                    geometryinformation_dict = meta_dict["geometryinformation"]
-                    res_obj.metadata.create_element('GeometryInformation',
-                                                    featureCount=geometryinformation_dict['featureCount'],
-                                                    geometryType=geometryinformation_dict['geometryType']
-                                                    )
+        else:  # all files have been removed, start it over
+            files_type = check_uploaded_files_type(files_list)
+            tmp_dir = None
+            uploaded_file_type = None
+            baseFilename = None
+            uploadedFileCount = 0
+            uploadedFilenameString = None
+            shp_full_path = None
+            validate_files_dict['are_files_valid'] = files_type['are_files_valid']
+            validate_files_dict['message'] = files_type['message']
 
-                else:
-                    validate_files_dict['are_files_valid'] = False
-                    validate_files_dict['message'] = 'Please upload valid file(s).'
+            if validate_files_dict['are_files_valid']:
+
+                if res_obj.metadata.originalfileinfo.all().first():
```

Comment:
@pkdash These if conditions have been there all along and are not newly added - see the removed code snippet above. Do you mean just to delete these metadata elements without checking whether they exist first since the delete() function handles the check already? Adding @zhiyuli into the loop as well.